examples/all-clusters-app/all-clusters-common/include/water-heater-mode.h

```diff
@@ -0,0 +1,73 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/mode-base-server/mode-base-server.h>
+#include <app/util/config.h>
+#include <cstring>
+#include <utility>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+namespace WaterHeaterMode {
+
+constexpr uint8_t ModeOff    = 0;
+constexpr uint8_t ModeManual = 1;
+constexpr uint8_t ModeTimed  = 2;
```

Comment:
```suggestion
constexpr uint8_t kModeOff    = 0;
constexpr uint8_t kModeManual = 1;
constexpr uint8_t kModeTimed  = 2;
```

and fixes to make use of this. Constant naming shoud be `kCamelCaseName`